### DISCUSSION TITLE
Direction for the official course for beginners

### DISCUSSION THREAD

**Discussion by thetutlage** [adonisjs/core#4205] [Polls]
We plan to launch an official course for beginners alongside the release of v6 and need your feedback to spend energy in the right direction.

We aim to help newcomers learn the basics of the framework and feel confident about using it.

> **Note**: We know that many framework users are super comfortable with AdonisJS and might be looking for advanced courses. However, we first want to focus on paving a path for someone just starting with AdonisJS.

Let's start with a poll. However, if you have deeper insights to share, feel free to comment.

## Terms

- **Project-based learning** means creating an end-to-end toy application and learning the framework concepts along the way. In this style, the reader will build an app by the end of the course, but they may not know about all the concepts, or their understanding of a particular concept can be very basic.
- **Topic-based learning** means learning about each topic in-depth. For example, if we are covering Routing, we can cover most of the routing APIs and some examples of where to use those APIs. But, no real/toy apps will be built during the course.
- **Written** course will be published as a website.
- **Screencasts** will be downloadable videos.

**Comment by adegbengaagoro** [comment]
I think Project based (hybrid) is better because it shows them how to use the framework. I think the documentation already works very well for "Written - topic based learning"

What I could consider the hybrid will be telling the viewers in the screen cast which sections of the documentation to deep-dive into based on the context of the given video and any supporting video tutorials that can give more context, we can keep the references to Adonis Mastery & Adocasts solely.

For example: if the screen cast is about building authentication, we link to:

* https://docs.adonisjs.com/guides/auth/introduction
* https://adocasts.com/topics/authentication
* https://adonismastery.com/courses/end-to-end-authentication

I believe this approach would provide a lot more value and adoption of the framework.

Cheers

**Reply by AlexandreGerault** [reply]
I also think that the "Written - Topic based" sounds a bit like what documentation already does well

However, I might not think of hybrid like you do 🤔 There are tutorials/teaching series where you actually learn on a project, but kinda topic based also. You have "authentication", "authorization" etc, but every time in a context that is closer to real context, no Foo/Bar/Baz stuff. I believe it's easier to learn when the application you're practicing on is the same all along the way and that you already know what's stuff it is suppose to solve.

**Comment by McSneaky** [comment]
I like @adegbengaagoro proposal of linking docs and external learning materials 👍 

One thing came to my mind is that Laracasts does it mostly topic based: https://laracasts.com/series/laravel-8-from-scratch and it's been working out real good

It sometimes builds on top of previous episod like: "In previous episode we made login page with Laravel Blade (Adonis Edge), now we will continue with server-side authentication" but overall are kinda topic based

**Reply by thetutlage** [reply]
Isn't this project based learning? Because, if I look the Blade section for example, Jeffrey covers just the topics needed to build that toy blog application? 

In case of topic based learning, I will expect him to cover everything related to Blade templates. Which might include components, passing props, slots or any other advanced uses cases of the same.

**Reply by McSneaky** [reply]
Aah indeed, he's been doing project based now. Been ages since I last watched "Laravel from scratch" episodes 😅

**Comment by ejazkhan10** [comment]
Project based learning is ideal for people that want to see how well adonisjs can be used to build something solid. I would also like to see 3rd party goodies like turbo-stream and turbo-frames integrated in the project.

**Reply by danielbergholz** [reply]
Having a default solution for the frontend like Hotwire would be really nice 👀

**Comment by termon** [comment]
I think project based learning provides a feedback loop for the learner, allowing them  to see a reasonably complex application develop incrementally over the duration of the course. This will give them the confidence to be able to tackle more advanced topics as ways of delivering more sophisticated functionality.

**Comment by KhanAmaan55** [comment]
IMO, 'Screencasts - Topic-based learning' is something that is needed as a beginner, when I started with Adonis js I learned more from documentation but always was confused with the process and all the cases of the topic, I have to search till the last of the documentation to even find a simple working and always wished for a tutorial that explains me the things and show me how it actually works.
And I think doc already works fine for  "Written - Topic-based learning" and "project-based learning" may not cover all the cases possible with a particular topic 
So, if you are planning to make the course for beginners, as a beginner I would suggest Screencasts - Topic-based learning

**Comment by jochenstu** [comment]
What I am most interested in is any very clearly positioned anti-status-quo pitch. In an increasingly everything react world, make it obvious what is great about moving into adonisjs and be clear about how to handle things that react does well.

You made an effort to rebuilt this remix app a while ago and I thought that was great and very instructional in that sense.

**Comment by vspinellis** [comment]
For me, when we create projects, we end up limiting a lot to that context. The Adonis documentation is already excellent, but it could be improved with more day-to-day examples focused on this topic, practical examples and best practices. I lack clear examples on Re-using controllers(service objects) for example. Projects you can do on their official youtube channel.

**Comment by thetutlage** [comment]
Alright, so here's the plan for the first official course. This course will not be able to cover all the framework topics, but we can always have more courses with time.

## Format of the course

Build an app from scratch and learn different topics of the framework we need to create the app.

## What's the app?

The app is called OS Hunt. It is a tutorial-friendly version of Product Hunt, but it is for sharing and exploring Open-source projects. The course outline will give an idea about the app's features. Here's a design screenshot.

![os-hunt-preview](https://github.com/adonisjs/core/assets/1706381/8c100681-41f4-4ea6-a2c8-f1252dcc80e9)

## Course outline

I have the course outline more or less finalized. Things might change as I hit recording, but not too much. Feedback welcomed.

![course-outline](https://github.com/adonisjs/core/assets/1706381/c6412ca5-b7bd-4ac9-a149-5bf8e286ecb5)
